src/engine/resource_manager/loader/curve.rs

```diff
@@ -10,15 +11,14 @@ use crate::{
 pub struct CurveLoader;
 
 impl ResourceLoader<CurveResource, CurveImportOptions> for CurveLoader {
-    type Output = BoxedLoaderFuture;
-
     fn load(
-        &mut self,
+        &self,
```

Comment:
Please leave this as `&mut self`, loader potentially can mutate its state during the load.

Reply:
You absolutely right, I just didn't thought about this, sorry.

Reply:
I'd have not thought of it prior trying to implement the sample wrapping loaders :).